nixos/modules/services/databases/lldap.nix

```diff
@@ -107,8 +107,21 @@ in
       wants = [ "network-online.target" ];
       after = [ "network-online.target" ];
       wantedBy = [ "multi-user.target" ];
+      # lldap defaults to a hardcoded `jwt_secret` value if none is provided, which is bad, because
+      # an attacker could create a valid admin jwt access token fairly trivially.
+      # Because there are 3 different ways `jwt_secret` can be provided, we check if any one of them is present,
+      # and if not, bootstrap a secret in `/var/lib/lldap/jwt_secret_file` and give that to lldap.
+      script = lib.optionalString (!cfg.settings ? jwt_secret) ''
+        if [[ -z "$LLDAP_JWT_SECRET_FILE" ]] && [[ -z "$LLDAP_JWT_SECRET" ]]; then
+          if [[ ! -e "./jwt_secret_file" ]]; then
+            ${lib.getExe pkgs.openssl} rand -base64 -out ./jwt_secret_file 32
```

Comment:
Is this running with custom umask at the service level? If not I think it will create jwt_secret_file world readable... Perhaps do `(umask 027; openssl ...)`?

Reply:
systemd's `DynamicUser=true` does some heavily lifting for us here.

`/var/lib/lldap` is just a symlink to `/var/lib/private/lldap`, while `/var/lib/private` has `0700/drwx------` perms, making it inaccessible to any unprivileged users.

But given there is also #287873, this is a really good and important catch nonetheless -- will fix. Thank you!

Also went ahead and checked what the perms of other files in `/var/lib/lldap` are (`nixosTests.lldap`).
```
$ ls -lash /var/lib/lldap/
total 128K
4.0K drwxr-xr-x 2 lldap lldap 4.0K Mar 10 16:57 .
4.0K drwx------ 3 root  root  4.0K Mar 10 16:57 ..
4.0K -rw-r--r-- 1 lldap lldap   45 Mar 10 16:57 jwt_secret_file
4.0K -r-------- 1 lldap lldap  128 Mar 10 16:57 server_key
112K -rw-r--r-- 1 lldap lldap 108K Mar 10 16:57 users.db
```

That's not great.
I would argue the `users.db` shouldn't be `0644/-rw-r--r--`.

Given this, I prefer service level `UMask=027` so the `umask` not only applies to `jwt_secret_file` but also to `users.db` and potentially other files as well.

And for instances that already have `users.db` set as is, I think changing `StateDirectoryMode=` from its default `0755` to `0750` should do the job.